src/rgw/rgw_http_client.cc

```diff
@@ -347,7 +347,7 @@ int RGWHTTPClient::init_request(const char *method, const char *url, rgw_http_re
   }
   curl_easy_setopt(easy_handle, CURLOPT_READFUNCTION, send_http_data);
   curl_easy_setopt(easy_handle, CURLOPT_READDATA, (void *)req_data);
-  if (is_upload_request(method)) {
+  if (send_data_hint || is_upload_request(method)) {
```

Comment:
does this mean we're sending a body with GET requests? is it possible to use PUT/POST instead, where `is_upload_request()` will do the right thing?